cmd/immudb/command/root.go

```diff
@@ -59,24 +59,35 @@ Setting the logging level and other options through environment variables:
 func (cl *Commandline) Immudb(immudbServer server.ImmuServerIf) func(*cobra.Command, []string) error {
 	return func(cmd *cobra.Command, args []string) (err error) {
 		var options *server.Options
+
 		if options, err = parseOptions(); err != nil {
 			return err
 		}
+
 		immudbServer := immudbServer.WithOptions(options)
-		if options.Logfile != "" {
-			if flogger, file, err := logger.NewFileLogger("immudb ", options.Logfile); err == nil {
-				defer file.Close()
-				immudbServer.WithLogger(flogger)
-			} else {
-				c.QuitToStdErr(err)
-			}
+
+		// initialize logger for immudb
+		flogger, file, err := logger.NewLogger(&logger.Options{
```

Comment:
I'm wondering if it's safe to remove the conditional `if options.Logfile != "" {`

Reply:
I've added that check inside this factory, `NewLogger` because there were too many ifs. If there is a json logger without file, there was a need to add an if else, and then a json logger with file would have a separate condition.

The problem is that the way immudb server is initialised, it sets the `NewSimpleLogger` by default in the `DefaultServer`. I see your point, the flogger can reset the logger even though it is defined, but I wanted to move to a single place where the logger is defined. WDYT?